### PR TITLE
ENH: copy keyword to set_axis

### PR DESCRIPTION
**Pull request by jbrockmendel**
This also opens the option of deprecating `inplace` here.

- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@Dr-Irv im not clear on what mypy is asking for.  help?

**Comment by Dr-Irv** [comment]
> @Dr-Irv im not clear on what mypy is asking for. help?

I think the issue is as follows.  In `core/generic.py`, you have the `inplace` argument on the 3rd overload typed as:
```python
inplace: bool_t | lib.NoDefault = ...,
```
But in `series.pyi` and `frame.pyi`, the 3rd overload has:
```python
inplace: bool = ...
```

You have to make sure the signatures are the same in the superclass and subclass for each overload, and that the declared method (without `@overload`) has all the possible types of each argument that are specified across all overloads.

File: pandas/tests/frame/methods/test_set_axis.py [hunk 1]
```diff
@@ -24,6 +24,55 @@ def test_set_axis(self, obj):
         result = obj.set_axis(new_index, axis=0, inplace=False)
         tm.assert_equal(expected, result)
 
+    def test_set_axis_copy(self, obj):
```
Comment (mroeschke) [inline pandas/tests/frame/methods/test_set_axis.py:27]: Does this test `inplace=True/copy=False`?

Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_set_axis.py:27]: yes

**Comment by jbrockmendel** [comment]
green, booyah

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**